src/main/java/mate/jdbc/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,131 @@
+                             Statement.RETURN_GENERATED_KEYS);) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(CANT_CREATE_MESSAGE, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateManufacturerStatement =
+                         connection.prepareStatement(UPDATE_MANUFACTURER_REQUEST,
+                             Statement.RETURN_GENERATED_KEYS);) {
+            updateManufacturerStatement.setString(1, manufacturer.getName());
+            updateManufacturerStatement.setString(2, manufacturer.getCountry());
+            updateManufacturerStatement.setObject(3, manufacturer.getId());
+            updateManufacturerStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException(CANT_UPDATE_MESSAGE, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteManufacturerStatement =
+                         connection.prepareStatement(DELETE_MANUFACTURER_REQUEST,
+                             Statement.RETURN_GENERATED_KEYS);) {
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() >= 1;
+        } catch (SQLException e) {
+            throw new RuntimeException(CANT_DELETE_MESSAGE, e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        Manufacturer manufacturer = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getManufacturerStatement =
+                         connection.prepareStatement(GET_MANUFACTURER_REQUEST,
+                             Statement.RETURN_GENERATED_KEYS);) {
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturer = new Manufacturer();
+                String name = resultSet.getString(NAME);
+                String country = resultSet.getString(COUNTRY);
+                manufacturer.setId(id);
+                manufacturer.setName(name);
+                manufacturer.setCountry(country);
+            }
```

Comment:
same for this part, if -> while; and separate method

Reply:
fixed